node/src/accountant/scanners.rs

```diff
@@ -1243,32 +1497,185 @@ mod tests {
+               0x000000000000000000000000000000000000000000000000000000000000223d."));
+        log_handler.exists_no_log_containing(&format!(
+            "DEBUG: {test_name}: Deleting an existing backup for a failed transaction {:?}",
+            hash_1
+        ));
+    }
+
+    fn common_body_for_failing_to_mark_rowids_tests(
+        test_name: &str,
+        pending_payable_dao: PendingPayableDaoMock,
+        hash_1: H256,
+        hash_2: H256,
+    ) {
+        let payable_1 = PendingPayable::new(make_wallet("blah111"), hash_1);
+        let payable_2 = PendingPayable::new(make_wallet("blah222"), hash_2);
+        let payable_dao = PayableDaoMock::new().mark_pending_payables_rowids_result(Err(
+            PayableDaoError::SignConversion(9999999999999),
+        ));
+        let mut subject = PayableScannerBuilder::new()
+            .payable_dao(payable_dao)
+            .pending_payable_dao(pending_payable_dao)
+            .build();
+        let sent_payables = SentPayables {
+            payment_procedure_result: Ok(vec![Correct(payable_1), Correct(payable_2)]),
+            response_skeleton_opt: None,
+        };
+
+        let caught_panic = catch_unwind(AssertUnwindSafe(|| {
+            subject.finish_scan(sent_payables, &Logger::new(test_name))
+        }))
+        .err()
+        .unwrap();
+
+        let panic_msg = caught_panic.downcast_ref::<String>().unwrap();
+        assert_eq!(
+            panic_msg,
+            "Unable to create a mark in the payable table for wallets 0x00000000000\
+        000000000000000626c6168313131, 0x00000000000000000000000000626c6168323232 due to \
+         SignConversion(9999999999999)"
+        );
+    }
+
+    #[test]
+    fn payable_scanner_fails_on_marking_pending_payable_and_panics_clear_not_having_run_into_nonexistent_fingerprints(
+    ) {
+        init_test_logging();
+        let test_name = "payable_scanner_fails_on_marking_pending_payable_and_panics_clear_not_having_run_into_nonexistent_fingerprints";
+        let hash_1 = make_tx_hash(248);
+        let hash_2 = make_tx_hash(139);
+        let pending_payable_dao = PendingPayableDaoMock::default()
+            .fingerprints_rowids_result(vec![(Some(7879), hash_1), (Some(7881), hash_2)]);
+
+        common_body_for_failing_to_mark_rowids_tests(
+            test_name,
+            pending_payable_dao,
+            hash_1,
+            hash_2,
+        );
+
+        TestLogHandler::new().exists_no_log_matching(&format!(
+            "ERROR: {test_name}: Payable fingerprints for (\
+         .*) not found but should exist by now; system unreliable"
+        ));
+    }
+
+    #[test]
+    fn payable_scanner_fails_on_marking_pending_payable_and_panics_clear_while_also_having_run_into_nonexistent_fingerprints(
+    ) {
+        init_test_logging();
+        let test_name = "payable_scanner_fails_on_marking_pending_payable_and_panics_clear_while_also_having_run_into_nonexistent_fingerprints";
+        let hash_1 = make_tx_hash(248);
```

Comment:
In general, if the parameters to `make_tx_hash` were supplied in hexadecimal (like 0xF8 instead of 248), they'd be easier to match to the hashes in the log messages and such.